### PR TITLE
Integrating DistilGPT-2 into UAgent Ecosystem

### PR DESCRIPTION
**Pull request by gaurav19908**
This Integration example sends a request to distilgpt2 interface API to auto complete text (in this example, My name is Clara and I am).
Note: Added distilgpt2 in integrations folder.

### CODE REVIEW

File: integrations/distilgpt2/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+##### HuggingFaceAPI Token
+* Visit HuggingFace.
+* Sign up or log in.
+* Goto Profile -> Settings -> Access Tokens.
+* Copy the old token or create new token.
+
+Once you have token, create a .env file in the distilgpt2-integrations/src directory.
+```bash
+export HUGGIN_FACE_ACCESS_TOKEN="{HuggingFaceAPI Token}"
```
Comment (jrriehl) [inline integrations/distilgpt2/README.md:20]: Any particular reason for not using their official spelling here at wherever else this is used?
```suggestion
export HUGGING_FACE_ACCESS_TOKEN="{HuggingFaceAPI Token}"
```

File: integrations/distilgpt2/src/agents/distilgpt2_agent.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        if response.status_code != 200:
+            # ctx.logger.error(response.json())
+            await ctx.send(sender, Error(error=f"Error: {response.json().get('error')}"))
+            return
+        message = response.json()[0]
+    except Exception as ex:
+        await ctx.send(sender, Error(error=f"Sorry, I wasn't able to answer your request this time. Feel free to try again. Error detail: {ex}"))
+        return
+
+    # ctx.logger.info(f"Sending response back to user: {message}")
```
Comment (jrriehl) [inline integrations/distilgpt2/src/agents/distilgpt2_agent.py:55]: Delete?

File: integrations/distilgpt2/src/agents/distilgpt2_agent.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    # Send the response back to the sender
+    await ctx.send(sender, Data.parse_obj(message))
+    return message
+
+# Create an instance of Protocol with a label "Request"
+distilgpt2_agent = Protocol("Request")
+
+
+@distilgpt2_agent.on_message(model=Request)
```
Comment (jrriehl) [inline integrations/distilgpt2/src/agents/distilgpt2_agent.py:65]: If you include the possible replies here, your protocol manifest will be more descriptive.
```suggestion
@distilgpt2_agent.on_message(model=Request, replies={Data, Error})
```

**Review by jrriehl — COMMENTED** [review]
Very nice contribution, just a few minor comments.

File: integrations/distilgpt2/README.md (hunk 1 above)
Comment (gaurav19908) [reply integrations/distilgpt2/README.md:20]: it was not intentional, it was spelling mistake.

File: integrations/distilgpt2/README.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+```
+
+Once you hit enter, a request will be sent to the agent every 10 mins, and you will be able to see your results in the console!
+You need to look for the following output in the logs:
+```
+ Adding agent to Bureau: {agent_address}
+```
+Copy the {agent_address} value, and replace AI_MODEL_AGENT_ADDRESS with this value distilgpt2_user.py file
+
+This example sends a request to distilgpt2 interface API to auto complete (in this example, Can you please let us know more details about your). You can change text by changing the COMPLETE_THIS variable in distilgpt2_user.py file.
```
[the remark this answers is not in the document]
Comment (gaurav19908) [reply integrations/distilgpt2/README.md]: Changed to this ->

In the src/distilgpt2_user.py file, there's a variable COMPLETE_THIS. Change its value to customize the text you want auto-completed. Default value is:

COMPLETE_THIS = "My name is Clara and I am"
Now you can enjoy creating unique sentences with DistilGPT2!

File: integrations/distilgpt2/src/agents/distilgpt2_agent.py (hunk 2 above)
Comment (gaurav19908) [reply integrations/distilgpt2/src/agents/distilgpt2_agent.py]: deleted

File: integrations/distilgpt2/src/agents/distilgpt2_agent.py (hunk 3 above)
Comment (gaurav19908) [reply integrations/distilgpt2/src/agents/distilgpt2_agent.py]: fixed

File: integrations/distilgpt2/src/agents/distilgpt2_user.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+# Check and top up the agent's fund if low
+fund_agent_if_low(user.wallet.address())
+
+# Define a protocol for the user agent
+distilgpt2_user = Protocol("Request")
+
+# Define a function that sends a message to the AI model agent every 360 seconds
+
+
+@distilgpt2_user.on_interval(360)
```
[the remark this answers is not in the document]
Comment (gaurav19908) [reply integrations/distilgpt2/src/agents/distilgpt2_user.py]: fixed

**Comment by gaurav19908** [comment]
Hi @jrriehl 

made changes according to your useful feedback, please have a look.

**Review by jrriehl — APPROVED** [review]
LGTM

**Pull request merged**